src/tests/system/tests/test_ipa.py

```diff
@@ -127,6 +127,87 @@ def test_ipa__hostpublickeys_by_ip(client: Client, ipa: IPA, public_keys: list[s
         assert f"{ip} {key}" in result.stdout_lines, "Did not get expected public keys!"
 
 
+@pytest.mark.ticket(gh=7583)
+@pytest.mark.importance("low")
+@pytest.mark.parametrize("option, output", [(None, "[{hostname}]:{port} {key}"), ("-o", "[{hostname}] {key}")])
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hostpublickeys_by_name_with_port(client: Client, ipa: IPA, public_keys: list[str],
+                                               option: str | None, output: str):
+    """
+    :title: sss_ssh_knownhosts returns public keys by host name with port
+    :setup:
+        1. Create host with SSH key
+        2. Configure SSSD with SSH responder
+        3. Start SSSD
+    :steps:
+        1. Lookup SSH key
+    :expectedresults:
+        1. All public keys were printed
+    :customerscenario: False
+    """
+    hostname = f"ssh-host.{ipa.domain}"
+    ip = "10.255.251.10"
+    port = 3333
+
+    ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)
+    client.sssd.enable_responder("ssh")
+    client.sssd.start()
+    args = []
+    if option is not None:
+        args.append(option)
+    args.append(f"[{hostname}]:{port}")
+
+    result = client.sss_ssh_knownhosts(*args)
+    assert result.rc == 0, "Did not get OpenSSH known hosts public keys!"
+    assert len(public_keys) == len(result.stdout_lines), "Did not get expected number of public keys!"
+    for key in public_keys:
+        assert (f"{output}.format(hostname={hostname}, port= {port}, key={key})"
```

Comment:
No. The problem is not that space. It is the `f` formatter at the beginning of the string.

The string for the assert should be:
```
assert output.format(hostname={hostname}, port= {port}, key={key}) in result.stdout_lines
```

Reply:
You could use:
```
assert output.format(hostname=hostname, port=port, key=key) in result.stdout_lines
```
or instead of using the parameterization for the format, you could simply use an `if option is None:` to make the code simpler.

Reply:
Updated the code.